velox/exec/LocalPartition.cpp

```diff
@@ -438,6 +440,120 @@ void LocalPartition::copy(
   target->copyRanges(input.get(), ranges);
 }
 
+template <TypeKind KIND>
+uint64_t retainedSizeWithSharedStringBuffers(
```

Comment:
Wondering if there is a way to avoid duplicating the logic of BaseVector::retainedSize and also make this code unit-testable.

CC: @bikramSingh91 @pedroerp @kgpai